packages/mdc-textfield/foundation.js

```diff
@@ -150,4 +156,20 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
       disabled: false,
     };
   }
+
+  registerAutoCompleteHandler_(input) {
+    return new Promise((resolve, reject) => {
+      let hasKeyInteraction = false;
+
+      input.addEventListener('input', (e) => {
```

Comment:
Two things here:

1. We shouldn't be using `addEventListener` here directly on the input, since foundations should try to avoid using DOM API methods directly whenever possible. Instead, let's create a `registerInputKeyboardEventHandler(type, handler)` method similar to `registerInput{Focus,Blur}Handler` above that can accommodate for this, and then include it within the vanilla component. 

2. We need to clean up these handlers after they're used. I see this being done in two places: when the promise is resolved ( / callback is called, since we should change the method to be IE11-friendly), as well as in the foundation's `destroy()` method, in case the foundation is destroyed before the textfield is interacted with. In order to facilitate this let's make a complementary `deregisterInputKeyboardEventHandler(type, handler)` method, and then refactor the event handlers in this method as well as the `hasKeyInteraction` variable into instance properties. 

Since these are adapter changes, we also need to update the component to implement them, as well as the README to document them. Finally, since every adapter change is considered breaking, we have to remember to add a `BREAKING CHANGE:` footer to the final squash and merge commit.